src/compiler/checker.ts

```diff
@@ -16990,6 +16984,7 @@ namespace ts {
             if (!reference.flowNode || !couldBeUninitialized && !(declaredType.flags & TypeFlags.Narrowable)) {
                 return declaredType;
             }
+            flowInvocationCount++;

```

Comment:
Nothing ever decrements this - that intentional?

Reply:
Yes, it's just a monotonically increasing count that allows us to detect if the function was (recursively) invoked.